### PR TITLE
Websocket metrics

### PR DESCRIPTION
**Pull request by vietj**
@purplefox can you review this please ?

### CODE REVIEW

**Comment by vietj** [comment]
can you review it @purplefox ?

File: src/main/java/io/vertx/core/http/impl/ClientConnection.java [hunk 1]
```diff
@@ -80,8 +80,8 @@
   // Requests can be pipelined so we need a queue to keep track of requests
   private final Queue<HttpClientRequestImpl> requests = new ArrayDeque<>();
   private final Handler<Throwable> exceptionHandler;
-  private final HttpClientMetrics metrics;
   private final Object metric;
+  final HttpClientMetrics metrics;
```
Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/ClientConnection.java]: I am not a huge fan of package protected members. Any chance of making private and providing getters?

File: src/main/java/io/vertx/core/http/impl/HttpServerImpl.java [hunk 2]
```diff
@@ -637,6 +637,7 @@ private void handshake(FullHttpRequest request, Channel ch, ChannelHandlerContex
         ServerWebSocketImpl ws = new ServerWebSocketImpl(vertx, theURI.toString(), theURI.getPath(),
           theURI.getQuery(), new HeadersAdaptor(request.headers()), wsConn, shake.version() != WebSocketVersion.V00,
           connectRunnable, options.getMaxWebsocketFrameSize());
+        ws.metric = metrics.connected(wsConn.metric(), ws);
```
Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/HttpServerImpl.java]: Do we distinguish between client websockets connected and server websockets?

Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/HttpServerImpl.java]: I'm not sure to understand, can you be more precise ?

File: src/main/java/io/vertx/core/http/impl/ClientConnection.java (hunk 1 above)
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/ClientConnection.java]: no problems with that. (I don't mind package protected members when they are final)

File: src/main/java/io/vertx/core/http/impl/HttpServerImpl.java (hunk 2 above)
Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/HttpServerImpl.java]: My question is: What does the websocket "connected" metric represent? Does it represent only server websockets that have been connected at the server side, or client websockets connected at the client side? Or both?

**Comment by purplefox** [comment]
I assume the the documentation for the metrics needs to be updated too (list of metrics)?

https://github.com/vert-x3/vertx-dropwizard-metrics/blob/master/src/main/asciidoc/java/index.adoc

I'm thinking the above should really be in core, not in the dropwizard metrics proj?

**Comment by vietj** [comment]
so the dropwizard impl has not yet been pushed. I'm waiting for validation in vertx core.

the doc for dropwizard is really specific to dropwizard

File: src/main/java/io/vertx/core/http/impl/HttpServerImpl.java (hunk 2 above)
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/HttpServerImpl.java]: it represents a server web socket that connected.

there are two SPI, one for client (HttpClientMetrics.connected with WebSocket), one for server (HttpServerMetrics#connected with ServerWebSocket)

**Pull request closed without merging**